salt/utils/schedule.py

```diff
@@ -562,17 +562,20 @@ def handle_func(self, func, data):
                             )
                         )
 
-            if 'return_job' in data and not data['return_job']:
-                pass
-            else:
-                # Send back to master so the job is included in the job list
-                mret = ret.copy()
-                mret['jid'] = 'req'
-                channel = salt.transport.Channel.factory(self.opts, usage='salt_schedule')
-                load = {'cmd': '_return', 'id': self.opts['id']}
-                for key, value in mret.items():
-                    load[key] = value
-                channel.send(load)
+            # Only attempt to return data to the master
+            # if the scheduled job is running on a minion.
+            if '__role' in self.opts and self.opts['__role'] == 'minion':
```

Comment:
Sorry, I missed this.  Good question, the reason for this fix is that the master has no master_uri configured, so when it tries to return a job running on the master the exception pops up.  I'm not overly familiar with syndics, but given that they're salt masters I would think it would be the same issue as running a job on a regular master.